components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/OAuthInfo.kt

```diff
@@ -4,36 +4,20 @@
 
 package mozilla.components.service.fxa
 
-import com.sun.jna.Pointer
-import com.sun.jna.Structure
-
-import java.util.Arrays
-
-class OAuthInfo internal constructor(raw: Raw) {
-
-    @JvmField val accessToken: String?
-    @JvmField val keys: String?
-    @JvmField val scope: String?
-
-    @Suppress("VariableNaming")
-    class Raw(p: Pointer) : Structure(p) {
-        @JvmField var access_token: Pointer? = null
-        @JvmField var keys: Pointer? = null
-        @JvmField var scope: Pointer? = null
-
-        init {
-            read()
+import org.mozilla.fxaclient.internal.OAuthInfo as InternalOAuthInfo;
+
+data class OAuthInfo(
```

Comment:
Can we add some KDocs here using `@property` to explain what keys, scope etc. are?

Reply:
Sure. This also points out that we really should be decoding these. `scopes` is a space-separated list, and `keys` is a JSON object. I've done that in the next patch.